tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
+The following code illustrates how to create, save, and schedule the fibonacci job:
+
+```python
+from girder.utility.model_importer import ModelImporter
+
+...
+
+# get the job model singleton which uses the factory method createJob
+jobModel = ModelImporter.model('job', 'jobs')
+
+# Create the job
+job = jobModel.createJob(
+    title='<NAME OF THE JOB>',
+    type='<JOB TYPE>', handler='worker_handler',
+    user=getCurrentUser(), public=False, args=(20,), kwargs={},
+    otherFields={
+        'celeryTaskName': 'my_custom_package.tasks.fibonacci'
+    })
+
+# Save the job
+jobModel.save(job)
+
+# Schedule the job
+jobModel.scheduleJob(job)
+
+return job
+```
+
+*This is code you might see in a REST request handler that is dedicated to launching fibonacci jobs. See the [Jobs](http://girder.readthedocs.io/en/latest/plugins.html?highlight=schedule#jobs) documentation for more information about creating, saving and scheduling jobs.*
+
+In the above code we get the job model singleton which uses a [factory](https://en.wikipedia.org/wiki/Factory_method_pattern) method ```createJob(...)``` to create the job. In our example you should replace the ```title``` and ```type``` keyword arguments with values that make sense for your purposes. Note that ```type``` is a arbitrary string used to categorize jobs, it has no inherent semantic value. ```user``` should be an instance of a user model,  the job will 'belong' to this user. ```public``` effects whether or not only the ```user``` can see the job or not. ```args``` Is a list of arguments to apply the consumer-side task object too (In our example ```n``` in ```my_custom_package.tasks.fibonacci``` will be 20). ```kwargs``` is a list of keyword arguments to apply the consumer-side task object too (In our example none). ```handler``` in our case *must be* ```"worker_handler"``` or the Worker plugin handler will not execute and the Celery message will not be produced. Finally ```otherFields``` is a dictionary of fields that are specific to Worker plugin job models. 
+
+Currently the Worker plugin supports two ```otherFields```:
+
++ **celeryTaskName** should be the name of the task object. This is almost always the python importable path to the task as a string (unless you have specifically set the [name](http://docs.celeryproject.org/en/latest/userguide/tasks.html#names) when decorating the function). In our example this is the string "my_custom_package.tasks.fibonacci". If you do not specify a ```celeryTaskName``` in ```otherFields``` it will use "girder_worker.run" by default.
++ **celeryQueue** allows you to place the produces message on a different RabbitMQ queue than the default ```celery``` queue.
+
+For the final part of our example we use the jobModel singleton to save the job to the database and then schedule the job to execute. 
```

Comment:
This is a little hard to read as it jumps back to the "narrative" of the code block after delving into the details of the params.  Maybe better to remove this line and add it to the "narrative", then focus on the params to `createJob`.

```
In the above code we get the job model singleton which uses a factory method createJob(...) to create the job, then saves the job and schedules the job to execute.
```

Though I think all of that is pretty easily inferred from the code and comments and could be cut out.

You could start with something like

```
The params to the job model singleton `createJob` method are 
```